src/api/cvc4cpp.cpp

```diff
@@ -4177,17 +4161,34 @@ Term Solver::mkTerm(Kind kind) const
 }
 
 Term Solver::mkTerm(Kind kind, Term child) const
+{
+  return mkTermHelper(kind, child);
+}
+
+Term Solver::mkTermHelper(Kind kind, Term child) const
 {
   NodeManagerScope scope(getNodeManager());
   CVC4_API_SOLVER_TRY_CATCH_BEGIN;
   CVC4_API_ARG_CHECK_EXPECTED(!child.isNull(), child) << "non-null term";
   CVC4_API_SOLVER_CHECK_TERM(child);
   checkMkTerm(kind, 1);
-
-  Node res = getNodeManager()->mkNode(extToIntKind(kind), *child.d_node);
+  Node res;
+  switch (kind)
+  {
+    case SINGLETON:
+    {
+      TypeNode type = (*child.d_node).getType();
```

Comment:
Can you comment here that the type of the Node and the type of the Term should be the same? This relies on the fact we don't special case Term::getSort().